contrib/epee/include/net/http_client.h

```diff
@@ -209,21 +219,29 @@ namespace net_utils
 				for(const auto& field : additional_params)
 					add_field(req_buff, field);
 
-				for (unsigned sends = 0; sends < 2; ++sends)
 				{
-					const std::size_t initial_size = req_buff.size();
 					const auto auth = m_auth.get_auth_field(method, uri);
 					if (auth)
 						add_field(req_buff, *auth);
 
 					req_buff += "\r\n";
 					//--
 
-					bool res = m_net_client.send(req_buff, timeout);
-					CHECK_AND_ASSERT_MES(res, false, "HTTP_CLIENT: Failed to SEND");
-					if(body.size())
+					if (do_copy_body) // small body
+					{
+						// Copy headers + body together and potentially send one fewer packet
+						req_buff.append(body.data(), body.size());
```

Comment:
Why not change the client to offer a writev-style method? Then you can skip the copies.

Reply:
`writev` isn't portable, would require much more code changes, and probably wouldn't be worth the effort since we only have a maximum of two distinct buffers to send. Since we're only copying small buffers in this PR, I'm making the performance bet that the time spent copying this small buffer is dwarfed by the network time it takes for TCP to make a round trip.

Reply:
I think @iamamyth was suggesting a change to the `m_net_client` interface to accept N buffers. I haven't looked too deeply on this, the easiest is probably `span<const boost::string_ref>` where a temporary array gets created here on the stack.